gr-digital/lib/qa_header_format.cc

```diff
@@ -29,25 +29,28 @@
 #include <gnuradio/digital/header_format_default.h>
 #include <gnuradio/blocks/unpack_k_bits.h>
 #include <gnuradio/expj.h>
+#include <gnuradio/xoroshiro128p.h>
 #include <volk/volk.h>
 #include <boost/test/unit_test.hpp>
 #include <stdio.h>
 #include <cmath>
 
 BOOST_AUTO_TEST_CASE(test_default_format)
 {
-  static const int N = 4800;
+  static const int N = 4800; /* multiple of 8 for easy random generation */
+  uint64_t rng_state[2];
   int upper8 = (N >> 8) & 0xFF;
   int lower8 = N & 0xFF;
 
   std::string ac = "1010101010101010"; //0xAAAA
   unsigned char *data = (unsigned char*)volk_malloc(N*sizeof(unsigned char),
                                                     volk_get_alignment());
-  srand (time(NULL));
-  for(unsigned int i = 0; i < N; i++) {
-    data[i] = rand() % 256;
+  uint64_t *data_ptr = reinterpret_cast<uint64_t*>(data);
+  xoroshiro128p_seed(rng_state, 42);
+  for(unsigned int i = 0; i < N / 8; ++i) {
```

Comment:
Was gonna complain for cases N not being multiples of 8, maybe actually tell people N *has* to be a multiple of 8, not only that it's convenient.

Reply:
True. Will revise comment above.

Maybe I should be writing a helper function in the xoroshiro sources to fill a buffer with random bytes. hm.

Reply:
True. Will revise comment above.

Maybe I should be writing a helper function in the xoroshiro sources to fill a buffer with random bytes. hm.